core/src/stored/append.cc

```diff
@@ -180,60 +319,80 @@ bool DoAppendData(JobControlRecord* jcr, BareosSocket* bs, const char* what)
      * An incomplete job can start the file_index at any number.
      * otherwise, it must start at 1.
      */
-    if (jcr->rerunning && file_index > 0 && last_file_index == 0) {
-      goto fi_checked;
-    }
-    if (file_index > 0
-        && (file_index == last_file_index
-            || file_index == last_file_index + 1)) {
-      goto fi_checked;
+    if (!(jcr->rerunning && file_index > 0 && last_file_index == 0)
+        && !(file_index > 0
+             && (file_index == last_file_index
+                 || file_index == last_file_index + 1))) {
+      Jmsg3(jcr, M_FATAL, 0, _("FI=%d from %s not positive or sequential=%d\n"),
+            file_index, what, last_file_index);
+      PossibleIncompleteJob(jcr, last_file_index);
+      ok = false;
+      break;
     }
-    Jmsg3(jcr, M_FATAL, 0, _("FI=%d from %s not positive or sequential=%d\n"),
-          file_index, what, last_file_index);
-    PossibleIncompleteJob(jcr, last_file_index);
-    ok = false;
-    break;
 
-  fi_checked:
     if (file_index != last_file_index) {
-      jcr->JobFiles = file_index;
       last_file_index = file_index;
+      if (file_currently_processed.GetFileIndex() > 0) {
+        processed_files.push_back(std::move(file_currently_processed));
+      }
+      file_currently_processed.Initialize(file_index);
```

Comment:
the standard mandates that a moved-from object is "in a valid, but unspecified state" which means invoking `Initialize()` as a member-function raises a flag here.
It is probably nicer to have an explicit constructor that takes file_index and then do `file_currently_processed = FileProcessed{file_index}` which will move-assign.